decoder/vaapiDecoderJPEG_unittest.cpp

```diff
@@ -139,24 +403,41 @@ VAAPIDECODER_JPEG_TEST(Factory)
     doFactoryTest(mimeTypes);
 }
 
-VAAPIDECODER_JPEG_TEST(Decode_Simple)
+template <std::size_t N>
+void decodeJpeg(const std::tr1::array<uint8_t, N>& data, uint32_t fourcc)
 {
     VaapiDecoderJPEG decoder;
     VideoConfigBuffer config;
     VideoDecodeBuffer buffer;
 
     config.flag = 0;
 
-    buffer.data = const_cast<uint8_t*>(g_SimpleJPEG.data());
-    buffer.size = g_SimpleJPEG.size();
+    buffer.data = const_cast<uint8_t*>(data.data());
+    buffer.size = data.size();
     buffer.timeStamp = 0;
 
     ASSERT_EQ(YAMI_SUCCESS, decoder.start(&config));
     ASSERT_EQ(YAMI_DECODE_FORMAT_CHANGE, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
 
-    EXPECT_TRUE(decoder.getOutput());
+    SharedPtr<VideoFrame> output = decoder.getOutput();
+    ASSERT_TRUE(output);
+    EXPECT_EQ(fourcc, output->fourcc);
+}
+
+VAAPIDECODER_JPEG_TEST(Decode_Simple)
+{
+    decodeJpeg(g_SimpleJPEG, YAMI_FOURCC_444P);
+    decodeJpeg(g_JPEG420, YAMI_FOURCC_IMC3);
+    /* this will trigger bug in dirver,
+     * see https://bugs.freedesktop.org/show_bug.cgi?id=97633c for details
+     */
+    /*
+    decodeJpeg(g_JPEG422H, YAMI_FOURCC_422H);
+    decodeJpeg(g_JPEG422V, YAMI_FOURCC_422V);
```

Comment:
Cool, I did not know about `xxd`... command line tools are better and faster.  I will be using that for sure ;)

Perhaps we should add the ffmpeg generated data to the driver test suite.